### PR TITLE
[DNM][WIP]Gaurav/ksd 112 implementation of sso setup in ksd

### PR DESCRIPTION
**Pull request by gauravsitlani**
**Description of your changes:**
- Ceph Dashboard SSO implementation based on: https://github.com/koor-tech/koor/pull/15

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://docs.koor.tech/docs/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://docs.koor.tech/docs/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/koor-tech/koor/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 1]
```diff
@@ -296,6 +296,59 @@ type DashboardSpec struct {
...
+	// Private key that should be used by Ceph Dashboard for signing and encryption
+	// +optional
+	SPPrivateKey SecretKeyRef `json:"spPrivateKey,omitempty"`
+}
+
+type UserRef struct {
+	// Name of the user created
+	Name string `json:"username,omitempty"`
+	// Respective role of the user which is needed for the same
+	Role string `json:"role,omitempty"`
```
Comment (galexrt) [inline pkg/apis/ceph.rook.io/v1/types.go:335]: Let's make this a string list/ slice so multiple roles can be specified.

Comment (gauravsitlani) [reply pkg/apis/ceph.rook.io/v1/types.go:335]: @galexrt sure sounds good.

**Comment by galexrt** [comment]
> Why are there so many changes?

We have "reset" the `master` branch of the repo to finally be better able to rebase new incoming changes and fix issues with our previous changes. The PR needs to be rebased/ re-created by copying the new code/ changes to a fresh branch that uses the latest `master` state.

**Comment by gauravsitlani** [comment]
opened a new PR: https://github.com/koor-tech/koor/pull/31 with all the changes and SSO code copied. Closing this one.

**Pull request closed without merging**